apollo-ast/src/main/resources/apollo.graphqls

```diff
@@ -22,7 +22,8 @@ directive @nonnull(fields: String! = "") on OBJECT | FIELD
 # `keyFields`: a selection set containing fields used to compute the cache key of an object. Order is important.
 # `embeddedFields`: a selection set containing fields that shouldn't create a new cache Record and should be
 # embedded in their parent instead. Order is unimportant.
-directive @typePolicy(keyFields: String! = "", embeddedFields: String! = "") on OBJECT | INTERFACE | UNION
+# `connectionFields`: a selection set containing fields that should be treated as Relay Connection fields. Order is unimportant.
+directive @typePolicy(keyFields: String! = "", embeddedFields: String! = "", connectionFields: String! = "") on OBJECT | INTERFACE | UNION
```

Comment:
Wondering if you could use `@requiresOptIn` on `connectionFields` 🥫?

Reply:
We could do it as a purely 'indicative' thing? But it won't impact the generated code as it's on a directive argument definition

Reply:
indicative is better than nothing. + we could output a warning during codegen?